### PR TITLE
Feature/add ibeaconscanner mock

### PR DESCRIPTION
**Pull request by eliaslecomte**
- Create a ScanService interface.
- Move scanning for ibeacons from IBeaconScanner to ScanService.
- UI test by using a mock ScanService (so it can run on emulators without bluetooth).
- Update SonarQube configuration.
- Small code improvements & added documentation.
- Update dependencies & build tools.

### CODE REVIEW

File: app/src/androidTest/java/mobi/inthepocket/android/beacons/app/MainActivityTest.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    {
+        this.mainActivityTestHelper.inputUUID("invalid uuid");
+
+        this.mainActivityTestHelper.clickOnButton(R.id.button_start);
+
+        this.mainActivityTestHelper.assertUUIDIsWrongErrorShown();
+    }
+
+    @Test
+    public void onWrongMajorTest()
```
Comment (Cliffus) [inline app/src/androidTest/java/mobi/inthepocket/android/beacons/app/MainActivityTest.java:57]: misschien ook tests toevoegen voor valid major & minor waarden?

File: app/src/main/java/mobi/inthepocket/android/beacons/app/MainActivity.java [hunk 2]
```diff
@@ -136,7 +132,45 @@ private void beaconsEnabled(final boolean isEnabled)
...
         this.buttonStartScanning.setEnabled(isEnabled);
         this.buttonStopScanning.setEnabled(isEnabled);
-        this.buttonAddFiveMore.setEnabled(isEnabled);
+    }
+
+    private boolean isValid()
+    {
+        boolean isValid = true;
+
+        if (!UUIDUtils.isValidUUID(this.editTextUuid.getText().toString()))
```
Comment (Cliffus) [inline app/src/main/java/mobi/inthepocket/android/beacons/app/MainActivity.java]: this.editTextUuid.getText().toString() extraheren in een String?

**Review by Cliffus — APPROVED** [review]

File: app/src/androidTest/java/mobi/inthepocket/android/beacons/app/MainActivityTest.java (hunk 1 above)
Comment (eliaslecomte) [reply app/src/androidTest/java/mobi/inthepocket/android/beacons/app/MainActivityTest.java:57]: good idea!

**Comment by eliaslecomte** [comment]
Regarding the failed travis-ci test. Sometimes our running UI tests get killed due to insufficient memory or to long running. In chat with travis-ci to see if they can help us with that.

**Pull request merged**